### PR TITLE
Vox chicken foods: Vox Nuggets and Vox Drumsticks

### LINKED ISSUE #31906: Vox turned into chickens can't be made into nuggets
(WEB REPORT BY: theophrastus REMOTE: 206.221.180.138:7777)
# Revision
8e30fc8902d0de8c3422052a7b82fdb555af56bc

# Description
After the end of a round, a doctor turned a Vox trader into a chicken, chloraled him, and put him on the kitchen table. I then tried to run him through the food processor to turn him into nuggets

# Steps to Reproduce
>turn vox into chicken via genetics
>put vox chicken into food processor
>activate
>vox will pop out harmlessly

# What you Expected
Vox nuggies

# What Actually Happened
disappointment

### PR DESCRIPTION
**Pull request by Eneocho**
Now with new icons.
Closes #31906.
[consistency][oversight]
:cl:
 * rscadd: Vox chickens (the green ones) become nuggets if processed, just like regular chickens do.
 * rscadd: Vox nuggets can also be made in the microwave. Same recipe as regular nuggets, but with vox chicken meat.
 * rscadd: Vox drumstick buckets can be made the same way regular buckets are made, but with vox chicken meat.
 * rscadd: Vox chicken meat works for all regular chicken meat recipes.

### CODE REVIEW

**Comment by Theophrastus1** [comment]
Thank god, finally vox nuggies

**Comment by Optimism333** [comment]
My chicken nuggets are BLUE.

**Comment by Eneocho** [comment]
We could use a different sprite for the vox nuggets.

**Comment by Theophrastus1** [comment]
A whole slew of vox cooking, thank you for enabling my gunk

**Comment by drawsstuff** [comment]
Isn't this already a thing? Vox meat acts as a substitute for chicken in all chicken recipes when cooking. Make sure these changes don't conflict. You may want to keep the first change and discard the rest as they're already handled by the game itself and weren't the subject of the issue report.

**Comment by Eneocho** [comment]
Vox chicks (the monkeys) don't give vox meat, just chicken meat. This is fluff, for the most part. If I understand microwavecode, these vox chick meat should work for any chicken-based recipe still, just that it has there are drumsticks and nuggets with dedicated icons for it.

**Comment by SonixApache** [comment]
did you test it

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 1]
```diff
@@ -654,6 +654,37 @@
...
+	..()
+	for(var/i=1; i <= storage_slots; i++)
+		new /obj/item/weapon/reagent_containers/food/snacks/vox_chicken_drumstick(src)
+	return
+
+/obj/item/weapon/storage/fancy/food_box/vox_chicken_bucket/remove_from_storage(obj/item/W as obj, atom/new_location, var/force = 0, var/refresh = 1)
+	. = ..()
+	if(!contents.len)
+		new/obj/item/trash/chicken_bucket(get_turf(src.loc))
+		if(istype(src.loc,/mob/living/carbon))
```
Comment (DamianX) [inline code/game/objects/items/weapons/storage/fancy.dm]: this isn't necessary, `qdel` will take care of it.

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 2]
```diff
@@ -654,6 +654,37 @@
...
+	icon_type = "drumstick"
+	can_only_hold = list("/obj/item/weapon/reagent_containers/food/snacks/vox_chicken_drumstick")
+	starting_materials = list(MAT_CARDBOARD = 3750)
+	w_type=RECYK_MISC
+
+/obj/item/weapon/storage/fancy/food_box/vox_chicken_bucket/New()
+	..()
+	for(var/i=1; i <= storage_slots; i++)
+		new /obj/item/weapon/reagent_containers/food/snacks/vox_chicken_drumstick(src)
+	return
```
Comment (DamianX) [inline code/game/objects/items/weapons/storage/fancy.dm]: return at the end of the proc does nothing, remove it

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 3]
```diff
@@ -4362,6 +4362,35 @@
...
+	item_state = "kfc_bucket"
+	trash = /obj/item/trash/chicken_bucket
+	food_flags = FOOD_MEAT
+	filling_color = "#4A75F4"
+	base_crumb_chance = 3
+
+/obj/item/weapon/reagent_containers/food/snacks/vox_nuggets/New()
+	..()
+	reagents.add_reagent(NUTRIMENT, 6)
+	bitesize = 1
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: these have no reason to be in New, move it above in the type definition

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 4]
```diff
@@ -4362,6 +4362,35 @@
...
+	desc = "I can't stand cold food. Unlike you, I ain't never ate from a trash can."
+	icon_state = "chicken_drumstick"
+	food_flags = FOOD_MEAT
+	filling_color = "#4A75F4"
+	base_crumb_chance = 0
+
+/obj/item/weapon/reagent_containers/food/snacks/vox_chicken_drumstick/New()
+	..()
+	reagents.add_reagent(NUTRIMENT, 3)
+	bitesize = 1
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: this is the other one

File: code/modules/reagents/reagent_containers/food/snacks/meat.dm [hunk 5]
```diff
@@ -125,6 +125,16 @@
...
 
+/obj/item/weapon/reagent_containers/food/snacks/meat/rawchicken/raw_vox_chicken
+	name = "vox chicken meat"
+	desc = "Vox, man. No discussion."
+	icon_state = "raw_vox_chicken"
+
+/obj/item/weapon/reagent_containers/food/snacks/meat/rawchicken/raw_vox_chicken/New()
+	..()
+	reagents.add_reagent(NUTRIMENT, 3)
+	bitesize = 1
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/food/snacks/meat.dm]: and this one

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 1 above)
Comment (Eneocho) [reply code/game/objects/items/weapons/storage/fancy.dm]: It's literally copypasted from the original chicken bucket code.

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 2 above)
Comment (Eneocho) [reply code/game/objects/items/weapons/storage/fancy.dm]: Same as above.

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 3 above)
Comment (Eneocho) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: That's how a lot of foods do it, doesn't seem to matter, I did change it however.

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 4 above)
Comment (Eneocho) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: See above.

File: code/modules/reagents/reagent_containers/food/snacks/meat.dm (hunk 5 above)
Comment (Eneocho) [reply code/modules/reagents/reagent_containers/food/snacks/meat.dm]: That's how the chicken meat and some other meats do it, but changed as well.

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 2 above)
Comment (DamianX) [reply code/game/objects/items/weapons/storage/fancy.dm]: don't care

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 1 above)
Comment (DamianX) [reply code/game/objects/items/weapons/storage/fancy.dm]: don't care

Comment (Eneocho) [reply code/game/objects/items/weapons/storage/fancy.dm]: What do I delete then?

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 2 above)
Comment (Eneocho) [reply code/game/objects/items/weapons/storage/fancy.dm]: done

File: code/game/objects/items/weapons/storage/fancy.dm (hunk 1 above)
Comment (DamianX) [reply code/game/objects/items/weapons/storage/fancy.dm]: the if and everything indented under it

Comment (Eneocho) [reply code/game/objects/items/weapons/storage/fancy.dm]: there

**Pull request merged**